internal/nodes/avi_model_nodes.go

```diff
@@ -666,12 +676,21 @@ func (o *AviVsNode) ReplaceSniSSLRefInSNINode(newSslNode *AviTLSKeyCertNode, key
 	return
 }
 
-func (o *AviVsNode) CheckHttpPolNameNChecksum(httpNodeName string, checksum uint32) bool {
-	for _, http := range o.HttpPolicyRefs {
-		if http.Name == httpNodeName {
-			//Check if their checksums are same
-			if http.GetCheckSum() == checksum {
-				return false
+func (o *AviVsNode) CheckHttpPolNameNChecksum(httpPolName, hppMapName string, checksum uint32) bool {
+	hppMapFound := false
+	for i, http := range o.HttpPolicyRefs {
+		if http.Name == httpPolName {
+			for _, hppMap := range o.HttpPolicyRefs[i].HppMap {
+				if hppMap.Name == hppMapName {
+					hppMapFound = true
+					if http.GetCheckSum() == checksum {
+						return false
+					}
+				}
+				//checksum is not same so need to modify
+				if hppMapFound {
```

Comment:
I think we dont need the variable hppMapFound, we can add this in an else block after line no: 688